src/auth/Auth.ts

```diff
@@ -84,18 +87,25 @@ class Auth {
   /**
    * Updates the current session with a new access token. On error,
    */
-  public renewSession = (): void => {
-    this.auth0.checkSession({}, (err, authResult) => {
-      if (authResult && authResult.accessToken && authResult.idToken) {
-        this.setSession(authResult);
-      } else if (err) {
-        this.logout();
-        console.log(err);
-        alert(
-          `Could not get a new token (${err.error}: ${err.errorDescription}).`
-        );
-      }
-    });
+  /**
+   * Updates the current session with a new access token. On error,
+   */
+  public renewSession = async (): Promise<void> => {
+    return new Promise((resolve, reject) =>
+      this.auth0.checkSession({}, (err, authResult) => {
+        if (authResult && authResult.accessToken && authResult.idToken) {
+          this.setSession(authResult);
+          resolve();
+        } else if (err) {
+          this.logout();
+          console.log(err);
+          alert(
+            `Could not get a new token (${err.error}: ${err.errorDescription}).`
```

Comment:
Is it safe to print those errors? Or could that get a security issue because we give out unnecessary information?